src/iterators/SingleCellIteratorInterface.h

```diff
@@ -70,3 +70,25 @@ class SingleCellIteratorInterfaceImpl : public SingleCellIteratorInterface<Parti
 };
 }  // namespace internal
 }  // namespace autopas
+
+/**
+ * gets a static cell iterator from an iteratorwrapper of a cell.
+ * @param iter the iterator to be defined
+ * @param cell the cell for which the iterator should be get
+ * @param body the body to be executed with the static iterator
+ */
+#define WITH_STATIC_CELL_ITER(iter, cell, body)                                                                       \
+  auto __wrapper = cell.begin();                                                                                      \
+  auto __ptr = __wrapper.get();                                                                                       \
+  {                                                                                                                   \
+    if (auto __##iter##ptr = dynamic_cast<                                                                            \
+            internal::SingleCellIterator<Particle, typename std::remove_reference<decltype(cell)>::type> *>(__ptr)) { \
+      auto iter = *__##iter##ptr;                                                                                     \
+      body                                                                                                            \
+    } else if (auto __##iter##ptr = dynamic_cast<RMMParticleCellIterator<Particle> *>(__ptr)) {                       \
+      auto iter = *__##iter##ptr;                                                                                     \
+      body                                                                                                            \
```

Comment:
since both branches seem to be identical can they be merged?

Reply:
no idea what you mean, but will merge anyways :)